sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala

```diff
@@ -68,6 +69,48 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
   import hiveContext._
   import spark.implicits._
 
+  test("SPARK-22431: table ctas - illegal nested type") {
```

Comment:
Put all illegal cases together since they share the same logic except the sql statements?